license-report not building

With cabal-plan 0.7.6, the conditional dependency on Cabal-syntax is bumped from `^>=3.12.0.0` to `^>=3.14.2.0`.

In 3.14, Cabal-syntax adds definitions for `</>` and `<.>`, which conflicts with the filepath imports in `LicenseReport`.

```
src-exe/LicenseReport.hs:56:22: error: [GHC-87543]
Error:     Ambiguous occurrence ‘<.>’.
    It could refer to
       either ‘Distribution.Utils.Path.<.>’,
              imported from ‘Distribution.Utils.Path’ at src-exe/LicenseReport.hs:32:1-40,
           or ‘System.FilePath.<.>’,
              imported from ‘System.FilePath’ at src-exe/LicenseReport.hs:35:1-32
              (and originally defined in ‘System.FilePath.Posix’).
   |
56 |   guard (rest == pns <.> "cabal")
   |                      ^^^

src-exe/LicenseReport.hs:81:25: error: [GHC-87543]
Error:     Ambiguous occurrence ‘</>’.
    It could refer to
       either ‘Distribution.Utils.Path.</>’,
              imported from ‘Distribution.Utils.Path’ at src-exe/LicenseReport.hs:32:1-40,
           or ‘System.FilePath.</>’,
              imported from ‘System.FilePath’ at src-exe/LicenseReport.hs:35:1-32
              (and originally defined in ‘System.FilePath.Posix’).
   |
81 |   let docDir = storeDir </> T.unpack (dispPkgId compilerId) </> T.unpack uidt </> "share" </> "doc"
   |                         ^^^
```

I’m trying with a workaround now:
```
cabal install cabal-plan \
  --constraint 'cabal-plan < 0.7.6' \
  --flags='exe license-report' \
  --ghc-options='-Wwarn'
```

--- a/.github/workflows/haskell-ci.yml
+++ b/.github/workflows/haskell-ci.yml
@@ -8,9 +8,9 @@
 #
 # For more information, see https://github.com/haskell-CI/haskell-ci
 #
-# version: 0.19.20250604
+# version: 0.19.20250722
 #
-# REGENDATA ("0.19.20250604",["github","cabal.project"])
+# REGENDATA ("0.19.20250722",["github","cabal.project"])
 #
 name: Haskell-CI
 on:
@@ -192,7 +192,11 @@ jobs:
           touch cabal.project.local
           echo "packages: ${PKGDIR_cabal_plan}" >> cabal.project
           echo "package cabal-plan" >> cabal.project
-          echo "    ghc-options: -Werror=missing-methods" >> cabal.project
+          echo "    ghc-options: -Werror=missing-methods -Werror=missing-fields" >> cabal.project
+          if [ $((HCNUMVER >= 90400)) -ne 0 ] ; then echo "package cabal-plan" >> cabal.project ; fi
+          if [ $((HCNUMVER >= 90400)) -ne 0 ] ; then echo "    ghc-options: -Werror=unused-packages" >> cabal.project ; fi
+          if [ $((HCNUMVER >= 90000)) -ne 0 ] ; then echo "package cabal-plan" >> cabal.project ; fi
+          if [ $((HCNUMVER >= 90000)) -ne 0 ] ; then echo "    ghc-options: -Werror=incomplete-patterns -Werror=incomplete-uni-patterns" >> cabal.project ; fi
           cat >> cabal.project <<EOF
           allow-newer: cabal-install-parsers-0.6.1:base
           allow-newer: cabal-install-parsers-0.6.1:transformers
@@ -231,6 +235,15 @@ jobs:
         run: |
           rm -f cabal.project.local
           $CABAL v2-build $ARG_COMPILER --disable-tests --disable-benchmarks all
+      - name: prepare for constraint sets
+        run: |
+          rm -f cabal.project.local
+      - name: constraint set license-report
+        run: |
+          $CABAL v2-build $ARG_COMPILER --disable-tests --disable-benchmarks --constraint='cabal-plan +license-report' all --dry-run
+          cabal-plan topo | sort
+          $CABAL v2-build $ARG_COMPILER --disable-tests --disable-benchmarks --constraint='cabal-plan +license-report' --dependencies-only -j2 all
+          $CABAL v2-build $ARG_COMPILER --disable-tests --disable-benchmarks --constraint='cabal-plan +license-report' all
       - name: save cache
         if: always()
         uses: actions/cache/save@v4

--- a/cabal-plan.cabal
+++ b/cabal-plan.cabal
@@ -1,6 +1,6 @@
 cabal-version:      2.2
 name:               cabal-plan
-version:            0.7.6.0
+version:            0.7.6.1
 synopsis:           Library and utility for processing cabal's plan.json file
 description:
   This package provides a library (see "Cabal.Plan") for decoding @plan.json@ files as

--- a/cabal.haskell-ci
+++ b/cabal.haskell-ci
@@ -1,2 +1,5 @@
 branches:     master
 installed:    +all -Cabal -Cabal-syntax
+
+constraint-set license-report
+  constraints: cabal-plan +license-report

--- a/src-exe/LicenseReport.hs
+++ b/src-exe/LicenseReport.hs
@@ -21,7 +21,6 @@ import           Data.Functor.Identity                  (Identity (..))
 import           Data.Map                               (Map)
 import           Data.List                              (nub)
 import qualified Data.Map                               as Map
-import           Data.Semigroup
 import           Data.Set                               (Set)
 import qualified Data.Set                               as Set
 import qualified Data.Text                              as T
@@ -29,7 +28,7 @@ import qualified Data.Text.IO                           as T
 import qualified Data.Version                           as DV
 import           Distribution.PackageDescription
 import           Distribution.PackageDescription.Parsec
-import           Distribution.Utils.Path
+import           Distribution.Utils.Path                (getSymbolicPath)
 import           Distribution.Pretty
 import           System.Directory
 import           System.FilePath
@@ -117,7 +116,7 @@ generateLicenseReport mlicdir plan uid0 cn0 = do
         rootPkgUnitIds = Set.fromList (map uId rootPkgUnits)
 
         -- the component of interest
-        Just root@Unit { uPId = PkgId pn0 _ } = Map.lookup uid0 (pjUnits plan)
+        root@Unit { uPId = PkgId pn0 _ } = Map.findWithDefault (error $ "cannot find " ++ show uid0) uid0 (pjUnits plan)
 
         fwdDeps = planJsonIdGraph' plan
         revDeps = invertMap fwdDeps
@@ -131,7 +130,7 @@ generateLicenseReport mlicdir plan uid0 cn0 = do
 
     let printInfo :: UnitId -> IO ()
         printInfo uid = do
-          let Just u = Map.lookup uid (pjUnits plan)
+          let u = Map.findWithDefault (error $ "cannot find " ++ show uid) uid (pjUnits plan)
 
               PkgId (PkgName pn) pv = uPId u
               isB = uType u == UnitTypeBuiltin

